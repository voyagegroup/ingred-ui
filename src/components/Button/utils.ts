import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
