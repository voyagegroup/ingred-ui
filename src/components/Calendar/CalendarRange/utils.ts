import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
