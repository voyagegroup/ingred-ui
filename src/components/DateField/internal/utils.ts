export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
