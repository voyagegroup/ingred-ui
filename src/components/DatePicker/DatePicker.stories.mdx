import { Meta, Story, ArgsTable, Canvas } from "@storybook/addon-docs/blocks";
import moment from "moment";
import DatePicker from "./DatePicker";
import "react-dates/lib/css/_datepicker.css";

<Meta title="Components/Inputs/DatePicker" component={DatePicker} />

# DatePicker

The wrapper of [react-dates](https://github.com/airbnb/react-dates).

For more detail props, please see [it](https://github.com/airbnb/react-dates#singledatepicker).

<ArgsTable of={DatePicker} />

## When the display is strange

Please import css from `react-dates`.

```tsx
import "react-dates/lib/css/_datepicker.css";
```

## Samples

### Basic

<Canvas>
  <Story name="Basic">
    {() => {
      {
        /* MEMO: To be unaffected by "Localize" story. */
      }
      moment.locale("en");
      const [date, setDate] = React.useState(moment());
      const handleChangeDate = (date) => {
        setDate(date);
      };
      return (
        <div style={{ height: "400px" }}>
          <DatePicker date={date} onDateChange={handleChangeDate} />
        </div>
      );
    }}
  </Story>
</Canvas>

### Error

<Canvas>
  <Story name="Error">
    {() => <DatePicker date={moment()} error={true} />}
  </Story>
</Canvas>

### Localize

<Canvas>
  <Story name="Localize">
    {() => {
      moment.locale("ja", {
        weekdaysShort: ["日", "月", "火", "水", "木", "金", "土"],
      });
      const renderMonthText = (day) => day.format("YYYY年M月");
      const displayFormat = () => "YYYY/MM/DD";
      const [date, setDate] = React.useState(moment());
      const handleChangeDate = (date) => {
        setDate(date);
      };
      return (
        <div style={{ height: "400px" }}>
          <DatePicker
            date={date}
            displayFormat={displayFormat}
            renderMonthText={renderMonthText}
            onDateChange={handleChangeDate}
          />
        </div>
      );
    }}
  </Story>
</Canvas>
