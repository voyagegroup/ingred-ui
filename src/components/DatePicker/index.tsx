export { default } from "./DatePicker";
export type { DatePickerProps } from "./DatePicker";
