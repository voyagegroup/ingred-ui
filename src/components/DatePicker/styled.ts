import styled from "styled-components";

export const IconContainer = styled.button`
  cursor: pointer;
  position: absolute;
  top: ${({ theme }) => theme.spacing}px;
  right: ${({ theme }) => theme.spacing}px;
  border: none;
  background: none;
`;
