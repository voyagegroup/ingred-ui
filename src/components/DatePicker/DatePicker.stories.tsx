import { ArgsTable, Description, Stories, Title } from "@storybook/addon-docs";
import { StoryObj } from "@storybook/react";
import dayjs from "dayjs";
import React from "react";
import DatePicker from "./DatePicker";
import "dayjs/locale/ja";
import localeData from "dayjs/plugin/localeData";

export default {
  title: "Components/Inputs/DatePicker",
  component: DatePicker,
  parameters: {
    docs: {
      source: { language: "tsx" },
      page: () => (
        <>
          <Title />
          <Description
            markdown={[
              "The wrapper of [react-dates](https://github.com/airbnb/react-dates).",
              "",
              "For more detail props, please see [it](https://github.com/airbnb/react-dates#singledatepicker).",
            ].join("\n")}
          />
          <ArgsTable of={DatePicker} />
          <Description
            markdown={[
              "## When the display is strange",
              "",
              "Please import css from `react-dates`.",
              "",
              "```tsx",
              "",
              'import "react-dates/lib/css/_datepicker.css";',
              "```",
            ].join("\n")}
          />
          <Stories includePrimary title="Stories" />
        </>
      ),
    },
  },
};

export const Basic: StoryObj = {
  render: () => {
    dayjs.locale("en");
    const [date, setDate] = React.useState(dayjs());
    const handleChangeDate = (date: dayjs.Dayjs | null) => {
      if (date === null) {
        return;
      }
      setDate(date);
    };
    return (
      <div style={{ height: "400px" }}>
        <DatePicker date={date} onDateChange={handleChangeDate} />
      </div>
    );
  },
};

export const Error: StoryObj = {
  render: () => {
    return <DatePicker date={dayjs()} error={true} onDateChange={() => {}} />;
  },
};

export const Localize: StoryObj = {
  render: () => {
    dayjs.locale("ja");
    dayjs.extend(localeData);
    const renderMonthText = (day: dayjs.Dayjs) => day.format("YYYYå¹´Mæœˆ");
    const displayFormat = () => "YYYY/MM/DD";
    const [date, setDate] = React.useState(dayjs());
    const handleChangeDate = (date: dayjs.Dayjs | null) => {
      if (date === null) {
        return;
      }
      setDate(date);
    };
    return (
      <div style={{ height: "400px" }}>
        <DatePicker
          date={date}
          locale={"ja"}
          localeData={dayjs().localeData()}
          displayFormat={displayFormat}
          renderMonthText={renderMonthText}
          onDateChange={handleChangeDate}
        />
      </div>
    );
  },
};
