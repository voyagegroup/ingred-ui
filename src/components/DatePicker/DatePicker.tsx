import * as React from "react";
import * as Styled from "./styled";
import "react-dates/initialize";
import moment from "moment";
import {
  RenderMonthProps,
  SingleDatePicker,
  SingleDatePickerShape,
} from "react-dates";
import Icon from "../Icon";
import Spacer from "../Spacer";
import { useLocaleProps } from "../../hooks/useLocaleProps";

function isOutsideRange() {
  return false;
}

export type DatePickerProps = Partial<
  Omit<SingleDatePickerShape, "date" | "onFocusChange">
> &
  // MEMO: Add RenderMonthProps to pass type check.
  RenderMonthProps & {
    date: moment.Moment | null;
    onDateChange: (date: moment.Moment | null) => void;
    error?: boolean;
  };

const DatePicker = React.forwardRef<HTMLDivElement, DatePickerProps>(
  (inProps, ref) => {
    const props = useLocaleProps({ props: inProps, name: "DatePicker" });
    const { date, error = false, ...rest } = props;

    const [focused, setFocused] = React.useState<boolean>(false);
    const onFocusChange = ({ focused }: { focused: boolean }) => {
      setFocused(focused);
    };

    return (
      <Styled.Container ref={ref} error={error}>
        <SingleDatePicker
          id="datePicker"
          focused={focused}
          date={date}
          isOutsideRange={isOutsideRange}
          numberOfMonths={1}
          enableOutsideDays={true}
          daySize={41}
          weekDayFormat="ddd"
          hideKeyboardShortcutsPanel={true}
          navPrev={
            <Styled.NavPrev>
              <Spacer p={0.5}>
                <Icon name="arrow_left" size="md" />
              </Spacer>
            </Styled.NavPrev>
          }
          navNext={
            <Styled.NavNext>
              <Spacer p={0.5}>
                <Icon name="arrow_right" size="md" />
              </Spacer>
            </Styled.NavNext>
          }
          onFocusChange={onFocusChange}
          {...rest}
        />
      </Styled.Container>
    );
  },
);

export default DatePicker;
