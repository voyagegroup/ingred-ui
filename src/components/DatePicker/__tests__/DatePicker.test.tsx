import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import DatePicker from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("DatePicker component testing", () => {
  afterEach(cleanup);

  test("DatePicker", () => {
    const { asFragment } = renderWithThemeProvider(
      <DatePicker date={null} onDateChange={jest.fn()} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
