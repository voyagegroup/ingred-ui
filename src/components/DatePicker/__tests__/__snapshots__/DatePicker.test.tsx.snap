// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component testing DatePicker 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL hzVvNx"
  >
    <div
      class="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          class="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <div
            class="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-datePicker"
              aria-label="Date"
              autocomplete="off"
              class="DateInput_input DateInput_input_1"
              id="datePicker"
              name="datePicker"
              placeholder="Date"
              type="text"
              value=""
            />
            <p
              class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-datePicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
